demo/js/library.js

```diff
@@ -6,19 +6,38 @@ class Library {
         this.initializeBasicLibrary();
     }
 
+    getHighestBookId(){
+        let max = -1;
+        for (let x = 0; x < this.getNumberOfBook(); x++) {
```

Comment:
This would be way more efficient

Math.max.apply(Math, this._books.map(function(book) { return book.getId(); }))